From a4b2e2b9739262a2948c3ab07929d54d2473a6b9 Mon Sep 17 00:00:00 2001
From: Daniel Dai <daijy@hortonworks.com>
Date: Wed, 28 Oct 2015 21:24:42 -0700
Subject: [PATCH 0457/1164] CDH-36102: HIVE-12282: beeline - update command
 printing in verbose mode (Daniel Dai, reviewed by
 Thejas Nair, Lefty Leverenz)

Conflicts:

	beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java

Change-Id: I89c9d1d0bdbe9696ca0ccc26c075a3605eb46158
---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |   22 ++++++++++++++------
 .../apache/hive/beeline/TestBeelineArgParsing.java |    3 +--
 2 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index d40dae0..adb5171 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -144,6 +144,7 @@
 
   private static final String HIVE_VAR_PREFIX = "--hivevar";
   private static final String HIVE_CONF_PREFIX = "--hiveconf";
+  static final String PASSWD_MASK = "[passwd stripped]";
 
   private final Map<Object, Object> formats = map(new Object[] {
       "vertical", new VerticalOutputFormat(this),
@@ -694,12 +695,9 @@ int initArgs(String[] args) {
     */
 
     if (url != null) {
-      String com = "!connect "
-          + url + " "
-          + (user == null || user.length() == 0 ? "''" : user) + " "
-          + (pass == null || pass.length() == 0 ? "''" : pass) + " "
-          + (driver == null ? "" : driver);
-      debug("issuing: " + com);
+      String com = constructCmd(url, user, pass, driver, false);
+      String comForDebug = constructCmd(url, user, pass, driver, true);
+      debug("issuing: " + comForDebug);
       dispatch(com);
     }
 
@@ -722,6 +720,18 @@ int initArgs(String[] args) {
     return code;
   }
 
+  private String constructCmd(String url, String user, String pass, String driver, boolean stripPasswd) {
+    String com = "!connect "
+        + url + " "
+        + (user == null || user.length() == 0 ? "''" : user) + " ";
+    if (stripPasswd) {
+      com += PASSWD_MASK + " ";
+    } else {
+      com += (pass == null || pass.length() == 0 ? "''" : pass) + " ";
+    }
+    com += (driver == null ? "" : driver);
+    return com;
+  }
   /**
    * Obtains a password from the passed file path.
    */
diff --git a/beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java b/beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java
index e54fe32..42e21bc 100644
--- a/beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java
+++ b/beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java
@@ -20,7 +20,6 @@
 
 import java.util.ArrayList;
 import java.util.List;
-
 import java.io.File;
 import java.io.FileOutputStream;
 
@@ -59,7 +58,7 @@ public void testSimpleArgs() throws Exception {
     TestBeeline bl = new TestBeeline();
     String args[] = new String[] {"-u", "url", "-n", "name",
       "-p", "password", "-d", "driver", "-a", "authType"};
-    Assert.assertEquals(0, bl.initArgs(args));
+    org.junit.Assert.assertEquals(0, bl.initArgs(args));
     Assert.assertTrue(bl.connectArgs.equals("url name password driver"));
     Assert.assertTrue(bl.getOpts().getAuthType().equals("authType"));
   }
-- 
1.7.9.5

